packages/disputer/src/disputer.js

```diff
@@ -172,54 +175,33 @@ class Disputer {
     }
 
     for (const disputeableLiquidation of disputableLiquidationsWithPrices) {
-      // Create the transaction.
-      const dispute = this.financialContract.methods.dispute(disputeableLiquidation.id, disputeableLiquidation.sponsor);
+      const disputeArgs = [disputeableLiquidation.id, disputeableLiquidation.sponsor];
 
       this.logger.debug({
         at: "Disputer",
         message: "Disputing liquidation",
         liquidation: disputeableLiquidation
       });
-      try {
-        // Get successful transaction receipt and return value or error.
-        const transactionResult = await runTransaction({
-          transaction: dispute,
-          config: {
-            gasPrice: this.gasEstimator.getCurrentFastPrice(),
-            from: this.account,
-            nonce: await this.web3.eth.getTransactionCount(this.account)
-          }
-        });
-        const receipt = transactionResult.receipt;
-        const returnValue = transactionResult.returnValue.toString();
-        const logResult = {
-          tx: receipt.transactionHash,
-          sponsor: receipt.events.LiquidationDisputed.returnValues.sponsor,
-          liquidator: receipt.events.LiquidationDisputed.returnValues.liquidator,
-          id: receipt.events.LiquidationDisputed.returnValues.liquidationId,
-          disputeBondPaid: receipt.events.LiquidationDisputed.returnValues.disputeBondAmount
-        };
-        this.logger.info({
+
+      // Submit the dispute transaction. This will use the DSProxy if configured or will send the tx with the unlocked EOA.
+      const logResult = await this.proxyTransactionWrapper.submitDisputeTransaction(disputeArgs);
+
+      if (logResult instanceof Error || !logResult)
+        this.logger.error({
           at: "Disputer",
-          message: "Position has been disputed!👮‍♂️",
-          liquidation: disputeableLiquidation,
-          disputeResult: logResult,
-          totalPaid: returnValue
+          message:
+            logResult.type === "call"
+              ? "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute✋"
+              : "Failed to dispute liquidation🚨",
+          logResult
         });
-      } catch (error) {
-        const message =
-          error.type === "call"
-            ? "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute✋"
-            : "Failed to dispute liquidation🚨";
-        this.logger.error({
+      else
+        this.logger.info({
           at: "Disputer",
-          message,
-          disputer: this.account,
+          message: "Liquidation has been disputed!👮‍♂️",
           liquidation: disputeableLiquidation,
-          error
```

Comment:
is `error` now in the `logResult`?

Reply:
yup! the `logResult` will either be an `Error` or an actual log. if it's of type `Error` then it will produce the error log, else it will produce the valid info log.